src/Store.lua

```diff
@@ -70,9 +91,22 @@ end
 	Get the current state of the Store. Do not mutate this!
 ]]
 function Store:getState()
+	if self.isDispatching then
+		error(("You may not call store:getState() while the reducer is executing. " ..
+			"The reducer (%s) has already received the state as an argument. " ..
+			"Pass it down from the top reducer instead of reading it from the store."):format(tostring(self._reducer)))
```

Comment:
This is probably something that happens with some regularity when executing thunks, which get the store itself passed into them instead of just state; however, I believe the thunk middleware _wraps_ the base dispatch, so when we're executing the thunk, we likely still haven't called the inner dispatch yet. I'm not sure about this, though, so this is something to look out for.

Reply:
Good call. I researched upstream redux-thunk, and while we are passing in the store instead of de-structuring, the structure and behavior is the same. isDispatching guard logic is pulled directly from upstream redux. As such, barring any co-routine/spawn/thread activity (which would *break* the flux pattern), I think we're okay since upstream is okay.